### PR TITLE
Use slack mentions instead of email address for awarding

### PR DESCRIPTION
**Pull request by riamaria**
Jira: COM-679
Reviewer: you?

### CODE REVIEW

File: guildbot.schema.sql [hunk 1]
```diff
@@ -27,11 +27,11 @@ CREATE TABLE `scores` (
 CREATE TABLE `users` (
   `id` int(11) unsigned NOT NULL AUTO_INCREMENT,
   `name` varchar(45) NOT NULL,
-  `email` varchar(45) NOT NULL,
+  `slack_id` varchar(50) NOT NULL,
```
Comment (oopu) [inline guildbot.schema.sql]: What certainty do we have of this size always working?

File: lambda_function.py [hunk 2]
```diff
@@ -32,9 +32,25 @@ def lambda_handler(data, context):
...
-            get_user_stats(profile, channel_id)
+            get_user_stats(slack_event["user"], profile, channel_id)
         elif text.startswith("award "):
-            return award(text, profile, channel_id)
+            pattern = re.compile(r"award ([\w-]+) to (<@)?(\w+)")
+
+            m = pattern.match(text)
+            if m == None:
+                slack.send_message(channel_id, "Cannot award points. Check your parameters and try again.")
+                return "500 INTERNAL SERVER ERROR"
```
Comment (oopu) [inline lambda_function.py]: There's no HTTP package for generating this type of response? That doesn't even look like a valid HTTP response.

Comment (oopu) [reply lambda_function.py]: Discussed offline. This only goes to logs. Suggested changing what is outputted given that we know that this isn't actually an HTTP response.

File: guildbot.schema.sql (hunk 1 above)
Comment (oopu) [reply guildbot.schema.sql]: None. Meh. Leave it as-is.

File: lambda_function.py [hunk 3]
```diff
@@ -47,70 +70,65 @@ def get_overall_stats(channel_id):
...
-        return "429 TOO MANY REQUESTS"
-    elif score == "SAME PERSON":
-        slack.send_message(channel_id, "You cannot award yourself")
-        return "403 FORBIDDEN"
+    db.get_or_create_user(user["real_name"], slack_id)
+
+    try:
+        score = db.cite_rubric_to_user(rubric, slack_id, db_user)
+    except db.TooSoonError:
+        slack.send_message(channel_id, "Cannot cite " + rubric_name + "; they just got cited this not too long ago")
```
Comment (jefQuery) [inline lambda_function.py]: Cite will be a confusing word to most. ELI5 it 
"Could not use" + rubric_name + "as a reason to award + " + user["real_name"] + " points; they just got cited this not too long ago"
